packages/palette/src/elements/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { FC } from "react"
+import { Box, Flex } from "@artsy/palette"
+import styled, { css, keyframes } from "styled-components"
+import { zIndex } from "styled-system"
+import { FocusOn } from "react-focus-on"
+
+export interface DrawerProps {
+  open: boolean
+  anchor?: "left" | "right"
+  zIndex?: number
+  onClose?(): void
+}
+
+export const Drawer: FC<DrawerProps> = ({
+  children,
+  anchor = "right",
+  zIndex = DEFAULT_DRAWER_Z_INDEX,
+  open,
+  onClose,
+}) => {
+  if (!open) {
+    return null
+  }
+
+  return (
+    <Container zIndex={zIndex} anchor={anchor}>
+      <Focus onClickOutside={onClose}>
+        <Content
+          backgroundColor="white100"
+          height="100%"
+          width={["100%", "auto"]}
+          overflowX="hidden"
+          overflowY="scroll"
+          anchor={anchor}
+          zIndex={zIndex}
+        >
+          {children}
+        </Content>
+      </Focus>
+
+      <Overlay
+        backgroundColor="black100"
+        height="100%"
+        display={["none", "flex"]}
+        onClick={onClose}
+        data-testid="drawer-overlay"
+        width="inherit"
+      />
+    </Container>
+  )
+}
+
+const DEFAULT_DRAWER_Z_INDEX = 9999
```

Comment:
This should just be like 1

Reply:
I thought we'd follow a similar pattern as we have in the [BaseModal](https://github.com/artsy/palette/blob/main/packages/palette/src/elements/Modal/ModalBase.tsx#L53-L56), but I changed in https://github.com/artsy/palette/pull/1295/commits/657df35bf582513603b7d209058a22b6a3093a62

Reply:
Good observation that that comment probably needs more context and less snark! At the time I was migrating legacy modals and that was what the value was at the time.

Reply:
Ah, I should have guessed something came up during past migrations, but no worries from my side, default 1 seems to be reasonable 👍